ast/check.go

```diff
@@ -1213,6 +1216,9 @@ func processAnnotation(ss *SchemaSet, annot *SchemaAnnotation, rule *Rule, allow
 	var schema interface{}
 
 	if annot.Schema != nil {
+		if ss == nil {
+			return nil, nil, nil
+		}
```

Comment:
This will ignore any by-ref `schemas` annotations if no `--schema` flag was supplied to the executed OPA command.

This might be a source of confusion, especially if by-ref and inlined schemas are mixed in the source, as the developer might not realize schemas are ignored if the `--schema` flag was forgotten.
Alternatively, we could always fail if the by-ref schema wasn't found, also when no `--schema` flag was supplied. This would require us to add the `--schema` flag to at least the `test` command. But could also be seen as a bigger breaking change to the behavior of `--schema`.